src/xb-builder-node.c

```diff
@@ -297,6 +301,41 @@ xb_builder_node_parse_literal_text (XbBuilderNode *self, const gchar *text, gssi
 	return g_string_free (tmp, FALSE);
 }
 
+/**
+ * xb_builder_node_tokenize_text:
+ * @self: a #XbBuilderNode
+ *
+ * Tokenize text added with xb_builder_node_set_text().
+ *
+ * Since: 0.3.1
+ **/
+void
+xb_builder_node_tokenize_text (XbBuilderNode *self)
+{
+	XbBuilderNodePrivate *priv = GET_PRIVATE (self);
+	g_auto(GStrv) tokens = NULL;
+	g_auto(GStrv) ascii_tokens = NULL;
+
+	g_return_if_fail (XB_IS_BUILDER_NODE (self));
+
+	if (priv->text == NULL)
+		return;
+	tokens = g_str_tokenize_and_fold (priv->text, NULL, &ascii_tokens);
+	for (guint i = 0; tokens[i] != NULL; i++) {
+		if (!xb_string_token_valid (tokens[i]))
+			continue;
+		xb_builder_node_add_token (self, tokens[i]);
```

Comment:
If you expect roughly all the tokens to be valid, `xb_builder_node_add_token()` is going to be called a lot, and hence will need to expand the `GPtrArray` allocation quite a lot. Might be more performant to pre-allocate the right number of elements in the array.

Reply:
It’s also going to call `g_strdup()` on all of the tokens, which is unnecessary given that you could steal them out of the `GStrv`, in this case anyway.

Reply:
> Might be more performant to pre-allocate the right number of elements in the array

I think I've done something like that; no observable difference in compile speed but I guess it all adds up.